client/edge.go

```diff
@@ -53,12 +61,43 @@ type EdgeClient interface {
 
 // NewEdgeConnector create EdgeConnector with the given BrokerConfig for the given EdgeClient
 func NewEdgeConnector(cfg *BrokerConfig, ecl EdgeClient) (*EdgeConnector, error) {
+	var tlsConfig *tls.Config
+	var certificates []tls.Certificate
+	var caCertPool *x509.CertPool
+	if len(cfg.Cert) > 0 { // implies the key will also be non-empty after validation
```

Comment:
If it implies that the key is not empty, then the check should be: if len(cfg.Cert) > 0 || len(cfg.Key). In this it will be aligned with the suite connector: https://github.com/eclipse-kanto/suite-connector/blob/main/config/tls_config.go